File: quick sort/quick-sort.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

void quickSort(vector<double>& prices, int low, int high) {
    if (low < high) {
        // Choosing the pivot 
        double pivot = prices[high];
        int i = low - 1;

        for (int j = low; j < high; j++) {
            if (prices[j] <= pivot) {
                i++;
                swap(prices[i], prices[j]);
            }
        }
        swap(prices[i + 1], prices[high]);
        int pivotIndex = i + 1;

        quickSort(prices, low, pivotIndex - 1);  
        quickSort(prices, pivotIndex + 1, high); 
    }
}

int main() {
    vector<double> prices = {99.99, 49.95, 299.49, 19.95, 199.95, 129.99};

    quickSort(prices, 0, prices.size() - 1);

    cout << "Sorted Prices: ";
    for (double price : prices) {
        cout << price << " ";
    }
    return 0;
}

File: quick sort/quick-sort.py
def partition(prices, low, high):
    pivot = prices[high]
    i = low - 1

    for j in range(low, high):
        if prices[j] < pivot:
            i += 1
            prices[i], prices[j] = prices[j], prices[i]
    prices[i + 1], prices[high] = prices[high], prices[i + 1]
    return i + 1

def quick_sort(prices, low, high):
    if low < high:
        pi = partition(prices, low, high)
        quick_sort(prices, low, pi - 1)
        quick_sort(prices, pi + 1, high)

prices = [99.99, 49.95, 299.49, 19.95, 199.95, 129.99]
quick_sort(prices, 0, len(prices) - 1)

print("Sorted prices:", prices)
